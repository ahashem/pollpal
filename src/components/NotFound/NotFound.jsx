import React from 'react';
import { NavLink } from 'react-router-dom';

import { pathRoot } from '../../AppRouter';

const NotFound = () => (
  <div>
    <h1>
      404
    </h1>
    <h3>
      This page does not exist
    </h3>
    <NavLink
      to={pathRoot}
    >
      go back to Questions List ->
    </NavLink>
  </div>
);

export default NotFound;
