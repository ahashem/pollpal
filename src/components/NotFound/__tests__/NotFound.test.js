import React from 'react';
import { shallow } from 'enzyme/build';

import NotFound from '../NotFound';


describe('<NotFound /> Component', () => {
  it('should render children and match snapshot', () => {
    const rendered = shallow(<NotFound/>);

    expect(rendered).toMatchSnapshot();
  });
});
