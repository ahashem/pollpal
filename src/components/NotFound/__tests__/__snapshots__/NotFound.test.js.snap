// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Component should render children and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          404
        </h1>,
        <h3>
          This page does not exist
        </h3>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          go back to Questions List -&gt;
        </NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404",
        },
        "ref": null,
        "rendered": "404",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This page does not exist",
        },
        "ref": null,
        "rendered": "This page does not exist",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "go back to Questions List ->",
          "to": "/",
        },
        "ref": null,
        "rendered": "go back to Questions List ->",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            404
          </h1>,
          <h3>
            This page does not exist
          </h3>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            go back to Questions List -&gt;
          </NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404",
          },
          "ref": null,
          "rendered": "404",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This page does not exist",
          },
          "ref": null,
          "rendered": "This page does not exist",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "go back to Questions List ->",
            "to": "/",
          },
          "ref": null,
          "rendered": "go back to Questions List ->",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
